### PR TITLE
Fixed use of deprecated methods and compatibility with PHP 5.6

### PR DESCRIPTION
**Pull request by samtuke**
This patch is necessary to run on my server and development machine.

### CODE REVIEW

**Comment by bramley** [comment]
Re the first file, init.php, are you suggesting that the iconv processing is now not needed for any version of php, not just 5.6? 

There are also some other unrelated changes - I cannot see where $timezone comes from. There is a commented-out define in the config file, SYSTEM_TIMEZONE.

You might want to correct the title, "deprecated" not "depreciated"!

**Comment by samtuke** [comment]
Hey @bramley thanks for the review. Wasn't aware of the timezone setting in the config file - why is this commented out by defult (which cases errors at runtime)? I assumed it wasn't set and hence added it.

File: public_html/lists/admin/init.php [hunk 1]
```diff
@@ -17,17 +17,21 @@
   error_reporting(0);
 }
 
+// Set the timezone based on value from config file
+date_default_timezone_set($timezone);
```
Comment (michield) [inline public_html/lists/admin/init.php]: where does $timezone come from?

The TZ is set at https://github.com/samtuke/phplist3/blob/master/public_html/lists/admin/connect.php#L70

If you think there's a good reason to move it here, that's fine.

**Comment by michield** [comment]
Iconv is only used on the attempt to convert a remote page to UTF8 if it isn't. https://github.com/phpList/phplist3/blob/master/public_html/lists/admin/lib.php#L1069
That's actually somewhat untested code, so I'm fine to take it out and drop the iconv dependency. Would be nice to address though at some point. I think there's a mantis for it.

iconv used to be used on the subject as well, https://github.com/phpList/phplist3/blob/master/public_html/lists/admin/send_core.php#L640 but that is commented out now.

File: public_html/lists/admin/init.php (hunk 1 above)
Comment (bramley) [inline public_html/lists/admin/init.php]: It was use of the $timezone variable that I was querying, as it doesn't exist.

**Comment by bramley** [comment]
Using iconv is fine, it is just the iconv_set_encoding() function that is deprecated in php 5.6.
But I cannot find any explanation of what effect the function has so it is not clear whether those lines can be removed or whether they need to be surround by a test of the php version, which is what Joomla has done https://github.com/joomla/joomla-cms/pull/5194

**Comment by michield** [comment]
Hmm, yes, and the change in PHP is undocumented

https://bugs.php.net/bug.php?id=69114

**Comment by michield** [comment]
looks like copying Joomla would be the best solution: https://github.com/joomla/joomla-cms/commit/ebead8274863aeb3598f7ad302d5c670984c608b#diff-6899d92657785065c65f0c4cdf51c574R28

**Comment by michield** [comment]
https://github.com/phpList/phplist3/commit/00fe90148b4aa763218afdc9023225a694f27b36

**Pull request closed without merging**